distance-method/distance_method.py

```diff
@@ -0,0 +1,981 @@
+    target_proteomes_dir: string
+      path to directory holding proteomes for all target organisms
+    verbose: boolean, optional
+      output details about the running processes of this function
+
+    Returns
+    -------
+    gene_map: dictionary
+      "two-way" dictionary storing gene names as keys and their pseudo
+      names as values, and vica versa
+    ref_db: dictionary
+      dictionary storing FASTA label as key and sequence as value for the
+      reference databases
+    species: integer
+      the number of species in the reference databases
+    """
+    gene_map = {}
+    ref_db = {}
+    if verbose:
+        sys.stdout.write("Target organism\tNumber of genes\n")
+    # each file contains genes for species
+    species = 0
+    for _file in listdir(target_proteomes_dir):
+        if verbose:
+            sys.stdout.write("%s. %s\t" % (species+1, basename(_file)))
+        with open(join(target_proteomes_dir, _file), 'rb') as readfile:
+            gene = 0
+            for label, seq in parse_fasta(readfile):
+                label = label.split()[0]
+                ref_db[label] = seq
+                sudo_label = "%s_%s" % (species, gene)
+                gene_map[label] = sudo_label
+                gene_map[sudo_label] = label
+                gene += 1
+        if verbose:
+            sys.stdout.write("%s\n" % gene)
+        species += 1
+    return gene_map, ref_db, species
+
+
+def launch_blast(query_proteome_fp,
+                 ref_fp,
+                 working_dir,
+                 e_value=10e-20,
+                 threads=1,
+                 debug=False):
+    """ Launch BLASTp given a query proteome and a reference database of
+    proteomes.
+
+    Parameters
+    ----------
+    query_proteome_fp: string
+      filepath to query proteome
+    ref_fp: string
+      filepath to reference proteome
+    working_dir: string
+      working directory path
+    e_value: float, optional
+      the cutoff E-value for BLASTP results
+    threads: integer
+      number of threads to use for running BLASTP
+    debug: boolean
+      if True, run function in debug mode
+
+    Returns
+    -------
+    out_file_fp: string
+      filepath to tabular alignment file output by
+      BLASTP
+    """
+    db_file_fp = join(working_dir, "%s" % basename(ref_fp))
+    # build blast database
+    makeblastdb_command = ["makeblastdb",
+                           "-in", ref_fp,
+                           "-out", db_file_fp,
+                           "-dbtype", "prot"]
+    proc = subprocess.Popen(makeblastdb_command,
+                            stdout=subprocess.PIPE,
+                            stderr=subprocess.PIPE,
+                            close_fds=True)
+    proc.wait()
+    stdout, stderr = proc.communicate()
+    if (stderr and debug):
+        print "[DEBUG] %s\n" % stderr
+
+    # launch blast
+    out_file_fp = join(
+        working_dir, "%s.blast" % basename(splitext(query_proteome_fp)[0]))
```

Comment:
Didn't want the original extension in the name, but it does make the code look cleaner. Changed.